tests/test_internals.py

```diff
@@ -34,3 +31,26 @@ def _get_docpath(self, builder, docname):
     content = open(path).read()
 
     assert app.builder.format == 'latex'
+
+
+@pytest.mark.sphinx(
+    srcdir=srcdir,
+    confoverrides={
+        'hoverxref_domains': ['py'],
+        'hoverxref_intersphinx': ['python'],
+        'hoverxref_auto_ref': True,
+        'extensions': [
+            'sphinx.ext.intersphinx',
+            'hoverxref.extension',
+        ],
+    },
+)
+def test_disconnect_intersphinx_listener(app, status, warning):
+    """Confirm that disconnecting the ``missing-reference`` listener from ``sphinx.ext.intershinx`` is successful."""
+    app.build()
+    listeners = []
+    for listener in app.events.listeners.get('missing-reference'):
+        module_name = inspect.getmodule(listener.handler).__name__
+        if module_name.startswith('sphinx.ext.intersphinx'):
+            listeners.append((module_name, listener))
+    assert not listeners, f"Expected to find zero listeners but found: {listeners}"
```

Comment:
We shouldn't copy the exact code we want to test in the test itself. We should instead test the _original_ function for `missing-reference` is not present in `app.events.listener` and the correct one _is present_.

Pseudo-code for the test:

```python
from sphinx.ext.intersphinx._resolve import missing_reference as intersphinx_missing_reference
from hoverxref.extension import missing_reference

app.build()
assert EventListener(id=Mock.ANY, priority=Mock.ANY, handler=intersphinx_missing_reference) not in app.events.listeners
assert EventListener(id=Mock.ANY, priority=Mock.ANY, handler=missing_reference) in app.events.listeners
```